### DISCUSSION TITLE
Implement assign meshoperation funtion in electronics desktop version 2024R2

### DISCUSSION THREAD

**Discussion by aknsj** [ansys/pyaedt#5007] [Q&A] [answered]
Hi, I have done full process automation of simulation using pyeadt, which has really helped reduce a lot of repetitive work!
But recently I've run into a small problem. The issue is that when using version 0.9.10 of pyaedt and version 2024R2 of electronics desktop, running the code for assign meshoperation causes the electronics desktop software to suddenly close. The error reported is shown below.
![image](https://github.com/user-attachments/assets/134058b0-0392-4871-b951-8e7106252388)

First, the following shows the code to open icepak.
```
aedt_version = "2024.2"
non_graphical = False

try:
    ipk = pyaedt.Icepak(projectname=project_path, designname=ipk_designname,
                        specified_version=aedt_version,
                        new_desktop_session=True,
                        close_on_exit=False,
                        non_graphical=non_graphical)
except Exception as e:
    print("An error occurred while trying to initialize Icepak:")
    print(e)
    raise
```
Second, the following is the code to assign meshoperation.
```
from collections import OrderedDict
i = 0
for solid_name in ipk.modeler.solid_names:
    if solid_name.lower().startswith(("copper","plastic")):
        ipk.mesh.assign_mesh_level({solid_name:2})
        ipk.mesh.meshoperations[i].props.update(OrderedDict({
            "MaxLevel": "3",
            "MinLevel": "2",
            "Local Mesh Parameters Enabled": True,
            "X count": "2",
            "Y count": "2",
            "Z count": "2"
        }))
        ipk.mesh.meshoperations[i].props.pop("Level")
        ipk.mesh.meshoperations[i].update()
        i+=1
```
My requirement is to implement assign meshoperation funtion in electronics desktop version 2024R2.
I would be very happy to know how to modify the code or the pyaedt version to fulfill this requirement. Looking forward to your reply, thanks!

**Comment by ansys-satyajeet** [comment, marked as answer]
Hi @aknsj 

You can try the following

```python
memory_meshop = "memory1_meshOP"
ipk.mesh.assign_mesh_level(mesh_order={"MEMORY1_1": 0}, 
                           name=memory_meshop)
for item in ipk.mesh.meshoperations:
    if item.name.startswith(memory_meshop):
        item.props["Local Mesh Parameters Enabled"] = True
        item.props["Local Mesh Parameters Type"] = "Box Local Mesh Parameters"
        item.props["X Count"] = 4
        item.props["Y Count"] = 2
        item.props["Z Count"] = 4
        item.update()
```
Here `MEMORY1_1` is name of the object. You should see the local mesh parameters updated.

**Reply by aknsj** [reply]
Hi, I tried the code you provided and successfully implemented  assign meshoperation function, thanks!
